### PR TITLE
Update Buildimage template to use new K8s package repository

### PR DESCRIPTION
**Pull request by creydr**
As per title (https://redhat-internal.slack.com/archives/CD87JDUB0/p1709542708643019)

### CODE REVIEW

**Comment by creydr** [comment]
/hold
waiting for test result from @skonto

File: cmd/generate/BuildImageDockerfile.template [hunk 1]
```diff
@@ -5,11 +5,10 @@ FROM {{.builder}} as builder
 
 RUN echo "[kubernetes]" >> /etc/yum.repos.d/kubernetes.repo && \
     echo "name=Kubernetes" >> /etc/yum.repos.d/kubernetes.repo && \
-    echo "baseurl=https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64" >> /etc/yum.repos.d/kubernetes.repo && \
+    echo "baseurl=https://pkgs.k8s.io/core:/stable:/v1.28/rpm/" >> /etc/yum.repos.d/kubernetes.repo && \
     echo "enabled=1" >> /etc/yum.repos.d/kubernetes.repo && \
     echo "gpgcheck=1" >> /etc/yum.repos.d/kubernetes.repo && \
-    echo "repo_gpgcheck=0" >> /etc/yum.repos.d/kubernetes.repo && \
-    echo "gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg" >> /etc/yum.repos.d/kubernetes.repo
+    echo "gpgkey=https://pkgs.k8s.io/core:/stable:/v1.28/rpm/repodata/repomd.xml.key" >> /etc/yum.repos.d/kubernetes.repo
```
Comment (matzew) [inline cmd/generate/BuildImageDockerfile.template:11]: looks like you forgot the tests

Comment (creydr) [reply cmd/generate/BuildImageDockerfile.template:11]: Done

**Comment by skonto** [comment]
For ref here is the doc: https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-using-native-package-management.

```
cat <<EOF | sudo tee /etc/yum.repos.d/kubernetes.repo
[kubernetes]
name=Kubernetes
baseurl=https://pkgs.k8s.io/core:/stable:/v1.29/rpm/
enabled=1
gpgcheck=1
gpgkey=https://pkgs.k8s.io/core:/stable:/v1.29/rpm/repodata/repomd.xml.key
EOF
```

We will need to update this based on some max K8s version we support? Could we autogenerate that value?
Currently it is 1.28.

**Comment by creydr** [comment]
> For ref here is the doc: https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-using-native-package-management.
> 
> ```
> cat <<EOF | sudo tee /etc/yum.repos.d/kubernetes.repo
> [kubernetes]
> name=Kubernetes
> baseurl=https://pkgs.k8s.io/core:/stable:/v1.29/rpm/
> enabled=1
> gpgcheck=1
> gpgkey=https://pkgs.k8s.io/core:/stable:/v1.29/rpm/repodata/repomd.xml.key
> EOF
> ```
> 
> We will need to update this based on some max K8s version we support? Could we autogenerate that value or we don't care? Currently it is 1.28.

I took the 1.28 from https://kubernetes.io/blog/2023/08/15/pkgs-k8s-io-introduction/#how-to-migrate-rpm. But let me update it to 1.29.

**Comment by skonto** [comment]
One other thing is that it takes almost 20min to run the install command. Check for Serving:

```
$ docker build --no-cache -t test:test -f ./openshift/ci-operator/build-image/Dockerfile .
[+] Building 162.8s (5/11)                                                                                      
[+] Building 163.0s (5/11)                                                                                      
[+] Building 163.1s (5/11)                                                                                      
[+] Building 1397.4s (12/12) FINISHED                                                                           
 => [internal] load build definition from Dockerfile                                                       0.1s
 => => transferring dockerfile: 1.29kB                                                                     0.0s
 => [internal] load .dockerignore                                                                          0.1s
 => => transferring context: 2B                                                                            0.0s
 => [internal] load metadata for registry.ci.openshift.org/openshift/release:golang-1.21                   0.0s
 => CACHED [1/8] FROM registry.ci.openshift.org/openshift/release:golang-1.21                              0.0s
 => [2/8] RUN echo "[kubernetes]" >> /etc/yum.repos.d/kubernetes.repo &&     echo "name=Kubernetes" >> /e  0.8s
 => [3/8] RUN yum install -y kubectl httpd-tools                                                        1327.9s
 => [4/8] RUN curl https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 | bash              9.1s 
 => [5/8] RUN GOFLAGS='' go install github.com/mikefarah/yq/v3@latest                                     14.3s 
 => [6/8] RUN GOFLAGS='' go install knative.dev/test-infra/tools/kntest/cmd/kntest@latest                 35.5s 
 => [7/8] RUN rm -rf /go/.cache                                                                            0.9s
 => [8/8] RUN chmod g+rw /etc/passwd                                                                       0.5s
 => exporting to image                                                                                     8.1s
 => => exporting layers                                                                                    8.1s
 => => writing image sha256:a788c4db1a8a9b3f1a3d7cd7ad608f08c48b86d534871cb26933c78f8a1e5711               0.0s
 => => naming to docker.io/library/test:test 
```
Was that the case before?

**Comment by skonto** [comment]
All other PRs are merged, merging this one.

**Pull request merged**